src/gensys.jl

```diff
@@ -88,23 +89,28 @@ end
 #- gensys methods -#
 ## -------------- ##
 
+# Wrapper methods ensuring no side effects on input arguments
+gensys(Γ0, Γ1, c, ψ, π) = gensys!(copy(Γ0), copy(Γ1), copy(c), copy(ψ), copy(π))
+gensys(Γ0, Γ1, c, ψ, π, div) = gensys!(copy(Γ0), copy(Γ1), copy(c), copy(ψ), copy(π), div)
```

Comment:
That's a very good point - it looks like only Γ0 and Γ1 are changed within `gensys`, and there only by the call to `schurfact!`. I'll change that to the non-mutating `schurfact` (which itself will copy Γ0 and Γ1).